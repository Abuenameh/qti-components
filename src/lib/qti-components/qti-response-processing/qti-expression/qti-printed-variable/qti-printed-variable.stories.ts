import '../../../index';
import { html } from 'lit';
import type { Meta, StoryObj } from '@storybook/web-components';
import { QtiAssessmentItem } from '../../../index';
import { within } from '@storybook/testing-library';

type Story = StoryObj; // <Props>;

const meta: Meta = {
  component: 'qti-printed-variable'
};
export default meta;

export const Single: Story = {
  render: args => {
    return html` <qti-assessment-item data-testid="qti-assessment-item">
      <qti-outcome-declaration base-type="string" cardinality="single" identifier="OUTCOME">
        <qti-default-value>
          <qti-value>SINGLE</qti-value>
        </qti-default-value>
      </qti-outcome-declaration>

      <qti-item-body>
        <qti-printed-variable identifier="OUTCOME"></qti-printed-variable>
      </qti-item-body>
      <qti-response-processing>
        <qti-set-outcome-value identifier="OUTCOME">
          <qti-base-value base-type="identifier">UITSTOOT</qti-base-value>
        </qti-set-outcome-value>
      </qti-response-processing>
    </qti-assessment-item>`;
  },
  play: ({ canvasElement }) => {
    const canvas = within(canvasElement);
    (canvas.getByTestId('qti-assessment-item') as QtiAssessmentItem).processResponse();
  }
};

export const Multiple: Story = {
  render: args => {
    return html` <qti-assessment-item data-testid="qti-assessment-item">
      <qti-outcome-declaration base-type="string" cardinality="multiple" identifier="OUTCOME">
        <qti-default-value>
          <qti-value>MULTIPLE</qti-value>
        </qti-default-value>
      </qti-outcome-declaration>

      <qti-item-body>
        <qti-printed-variable identifier="OUTCOME"></qti-printed-variable>
      </qti-item-body>
      <qti-response-processing>
        <qti-set-outcome-value identifier="OUTCOME">
          <qti-base-value base-type="identifier">UITSTOOT</qti-base-value>
        </qti-set-outcome-value>
      </qti-response-processing>
    </qti-assessment-item>`;
  },
  play: ({ canvasElement }) => {
    const canvas = within(canvasElement);
    (canvas.getByTestId('qti-assessment-item') as QtiAssessmentItem).processResponse();
  }
};
