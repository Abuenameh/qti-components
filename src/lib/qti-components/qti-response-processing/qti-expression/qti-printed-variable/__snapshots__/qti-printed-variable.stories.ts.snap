// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`lib/qti-components/qti-response-processing/qti-expression/qti-printed-variable Multiple play-test 1`] = `
<div id="root-inner">
  <qti-assessment-item data-testid="qti-assessment-item">
    <qti-outcome-declaration base-type="string"
                             cardinality="multiple"
                             identifier="OUTCOME"
    >
      <qti-default-value>
        <qti-value>
          MULTIPLE
        </qti-value>
      </qti-default-value>
    </qti-outcome-declaration>
    <qti-item-body>
      <qti-printed-variable identifier="OUTCOME">
      </qti-printed-variable>
    </qti-item-body>
    <qti-response-processing>
      <qti-set-outcome-value identifier="OUTCOME">
        <qti-base-value base-type="identifier">
          UITSTOOT
        </qti-base-value>
      </qti-set-outcome-value>
    </qti-response-processing>
  </qti-assessment-item>
</div>
`;

exports[`lib/qti-components/qti-response-processing/qti-expression/qti-printed-variable Multiple play-test 2`] = `
<div id="root-inner">
  <qti-assessment-item data-testid="qti-assessment-item">
    <qti-outcome-declaration base-type="string"
                             cardinality="multiple"
                             identifier="OUTCOME"
    >
      <qti-default-value>
        <qti-value>
          MULTIPLE
        </qti-value>
      </qti-default-value>
    </qti-outcome-declaration>
    <qti-item-body>
      <qti-printed-variable identifier="OUTCOME">
      </qti-printed-variable>
    </qti-item-body>
    <qti-response-processing>
      <qti-set-outcome-value identifier="OUTCOME">
        <qti-base-value base-type="identifier">
          UITSTOOT
        </qti-base-value>
      </qti-set-outcome-value>
    </qti-response-processing>
  </qti-assessment-item>
</div>
`;

exports[`lib/qti-components/qti-response-processing/qti-expression/qti-printed-variable Single play-test 1`] = `
<div id="root-inner">
  <qti-assessment-item data-testid="qti-assessment-item">
    <qti-outcome-declaration base-type="string"
                             cardinality="single"
                             identifier="OUTCOME"
    >
      <qti-default-value>
        <qti-value>
          SINGLE
        </qti-value>
      </qti-default-value>
    </qti-outcome-declaration>
    <qti-item-body>
      <qti-printed-variable identifier="OUTCOME">
      </qti-printed-variable>
    </qti-item-body>
    <qti-response-processing>
      <qti-set-outcome-value identifier="OUTCOME">
        <qti-base-value base-type="identifier">
          UITSTOOT
        </qti-base-value>
      </qti-set-outcome-value>
    </qti-response-processing>
  </qti-assessment-item>
</div>
`;
